Code/Framework/AzFramework/Platform/Linux/AzFramework/Input/Devices/Keyboard/InputDeviceKeyboard_xcb.cpp

```diff
@@ -0,0 +1,265 @@
+                case XKB_KEY_KP_0: return &InputDeviceKeyboard::Key::NumPad0;
+                case XKB_KEY_KP_1: return &InputDeviceKeyboard::Key::NumPad1;
+                case XKB_KEY_KP_2: return &InputDeviceKeyboard::Key::NumPad2;
+                case XKB_KEY_KP_3: return &InputDeviceKeyboard::Key::NumPad3;
+                case XKB_KEY_KP_4: return &InputDeviceKeyboard::Key::NumPad4;
+                case XKB_KEY_KP_5: return &InputDeviceKeyboard::Key::NumPad5;
+                case XKB_KEY_KP_6: return &InputDeviceKeyboard::Key::NumPad6;
+                case XKB_KEY_KP_7: return &InputDeviceKeyboard::Key::NumPad7;
+                case XKB_KEY_KP_8: return &InputDeviceKeyboard::Key::NumPad8;
+                case XKB_KEY_KP_9: return &InputDeviceKeyboard::Key::NumPad9;
+                case XKB_KEY_KP_Add: return &InputDeviceKeyboard::Key::NumPadAdd;
+                case XKB_KEY_KP_Decimal: return &InputDeviceKeyboard::Key::NumPadDecimal;
+                case XKB_KEY_KP_Divide: return &InputDeviceKeyboard::Key::NumPadDivide;
+                case XKB_KEY_KP_Enter: return &InputDeviceKeyboard::Key::NumPadEnter;
+                case XKB_KEY_KP_Multiply: return &InputDeviceKeyboard::Key::NumPadMultiply;
+                case XKB_KEY_KP_Subtract: return &InputDeviceKeyboard::Key::NumPadSubtract;
+                case XKB_KEY_apostrophe: return &InputDeviceKeyboard::Key::PunctuationApostrophe;
+                case XKB_KEY_backslash: return &InputDeviceKeyboard::Key::PunctuationBackslash;
+                case XKB_KEY_bracketleft: return &InputDeviceKeyboard::Key::PunctuationBracketL;
+                case XKB_KEY_bracketright: return &InputDeviceKeyboard::Key::PunctuationBracketR;
+                case XKB_KEY_comma: return &InputDeviceKeyboard::Key::PunctuationComma;
+                case XKB_KEY_equal: return &InputDeviceKeyboard::Key::PunctuationEquals;
+                case XKB_KEY_hyphen: return &InputDeviceKeyboard::Key::PunctuationHyphen;
+                case XKB_KEY_period: return &InputDeviceKeyboard::Key::PunctuationPeriod;
+                case XKB_KEY_semicolon: return &InputDeviceKeyboard::Key::PunctuationSemicolon;
+                case XKB_KEY_slash: return &InputDeviceKeyboard::Key::PunctuationSlash;
+                case XKB_KEY_asciitilde: return &InputDeviceKeyboard::Key::PunctuationTilde;
+                case XKB_KEY_ISO_Group_Shift: return &InputDeviceKeyboard::Key::SupplementaryISO;
+                case XKB_KEY_Pause: return &InputDeviceKeyboard::Key::WindowsSystemPause;
+                case XKB_KEY_Print: return &InputDeviceKeyboard::Key::WindowsSystemPrint;
+                case XKB_KEY_Scroll_Lock: return &InputDeviceKeyboard::Key::WindowsSystemScrollLock;
+                default: return nullptr;
+            }
+        }
+
+        template<auto freeFn>
+        using DeleterForFreeFn = AZStd::integral_constant<decltype(freeFn), freeFn>;
```

Comment:
This is honestly an abuse of integral constant to both store the `type` of the function pointer and the function pointer in the `value` constexpr simultaneously.
It does work though and a pointer is technically an integer,

Reply:
The alternative is to define some struct with an `operator()` that takes the right type of arguments, would you prefer that?
XCB, being a C library, has lots of these deleter functions, I expect this pattern can be reused throughout the XCB platform support code.

Reply:
Well if you were to define a struct, it would the same as the integeral constant class anyway.
It would a templated class with an `operator()`, though that does better document the intent
```
template <auto DeleterFunction>
struct DeleterFunctionWrapper
{
    using value_type = decltype(DeleterFunction);
    static constexpr auto value = DeleterFunction;
    constexpr value_type operator()()
    {
        return value;
    }
};
template <typename T, auto DeleterFunction>
using XkbUniquePtr = AZStd::unique_ptr<T, DeleterFunctionWrapper<DeleterFunction>>;
```

Reply:
Since there's probably going to be more of these as the xcb support is expanded, I'm going to leave this for a later change.